src/freeze/disable.rs

```diff
@@ -0,0 +1,83 @@
+use super::*;
+
+pub struct DisableFreezeArgs {
+    pub keypair: Option<String>,
+    pub rpc_url: Option<String>,
+    pub cache: String,
+    pub candy_machine: Option<String>,
+}
+
+pub fn process_disable_freeze(args: DisableFreezeArgs) -> Result<()> {
+    let sugar_config = sugar_setup(args.keypair.clone(), args.rpc_url.clone())?;
+    let client = setup_client(&sugar_config)?;
+    let program = client.program(CANDY_MACHINE_ID);
+
+    // The candy machine id specified takes precedence over the one from the cache.
+    let candy_machine_id = match args.candy_machine {
+        Some(ref candy_machine_id) => candy_machine_id.to_owned(),
+        None => {
+            let cache = load_cache(&args.cache, false)?;
+            cache.program.candy_machine
+        }
+    };
+
+    let candy_pubkey = Pubkey::from_str(&candy_machine_id)
+        .map_err(|_| anyhow!("Failed to parse candy machine id: {}", &candy_machine_id))?;
+
+    println!(
+        "{} {}Loading candy machine",
+        style("[1/2]").bold().dim(),
+        LOOKING_GLASS_EMOJI
+    );
+    println!("{} {}", style("Candy machine ID:").bold(), candy_machine_id);
+
+    let pb = spinner_with_style();
+    pb.set_message("Connecting...");
+    let candy_machine_state =
+        get_candy_machine_state(&sugar_config, &Pubkey::from_str(&candy_machine_id)?)?;
+
+    pb.finish_with_message("Done");
+
+    assert_correct_authority(
+        &sugar_config.keypair.pubkey(),
+        &candy_machine_state.authority,
+    )?;
+
+    println!(
+        "\n{}{}{}{}Turning off freeze feature for candy machine",
+        style("[2/2]").bold().dim(),
+        FIRE_EMOJI,
+        RIGHT_ARROW_EMOJI,
+        ICE_CUBE_EMOJI
+    );
+
+    let pb = spinner_with_style();
+    pb.set_message("Sending remove freeze transaction...");
+
+    let signature = disable_freeze(&program, &candy_pubkey)?;
```

Comment:
Is it possible to check whether the candy machine has freeze enabled before trying to disable? Or perhaps improve the error message?
<img width="682" alt="image" src="https://user-images.githubusercontent.com/729235/190187275-188c3208-0ed0-448f-8b25-b4d75c1b7a2f.png">

Reply:
I am also not sure about the 3 emojis – I understand the concept, but they look a bit out of place compared to the other ones.

Reply:
[This commit](https://github.com/metaplex-foundation/sugar/pull/345/commits/6bdb79ff9f6e5b53a1014497116eb5ec45a2b478) fixes these two issues.

Reply:
It also changes the candy machine to be an option for consistency and so a user doesn't have to provide it if they provide a freeze time.

Reply:
Great!